src-built-in/components/toolbar/stores/toolbarStore.js

```diff
@@ -246,33 +245,169 @@ class _ToolbarStore {
-				FSBL.Clients.LauncherClient.bringWindowsToFront()
-			});
-			FSBL.Clients.HotkeyClient.addGlobalHotkey([keys.ctrl, keys.alt, keys.down], (err) => {
-				if (err) { FSBL.Clients.Logger.error(`HotkeyClient.addGlobalHotkey failed, error:`, err); }
-				FSBL.Clients.WorkspaceClient.minimizeAll()
-			});
-			FSBL.Clients.HotkeyClient.addGlobalHotkey([keys.ctrl, keys.alt, keys.f], (err) => {
-				if (err) { FSBL.Clients.Logger.error(`HotkeyClient.addGlobalHotkey failed, error:`, err); }
-				self.Store.setValue({ field: "searchActive", value: true });
-			});
+		/**
+		 * hotkeyActions
+		 * A string can be provided by a hotkey and the corresponding function gets returned.
+		 * ?This is the only way to be able to send actions from JSON.
+		 * ?This flexibility allows us to change the action via config.
+		 * @param {string} action
+		 */
+		const hotkeyActions = action => {
+			const self = this;
+			switch (action) {
+
+				case 'SHOW_TOOLBAR':
+					return function showToolbar() {
+						self.Store.setValue({ field: "searchActive", value: false });
+						self.showToolbarAtFront();
+					}
+
+				case 'HIDE_TOOLBAR':
+					return self.hideToolbar
+
+				case 'BRING_WINDOWS_TO_FRONT':
+					return function bringWindowsToFront() {
+						FSBL.Clients.LauncherClient.bringWindowsToFront()
+					}
+
+				case 'MINIMIZE_ALL':
+					return function minimizeAll() {
+						FSBL.Clients.WorkspaceClient.minimizeAll()
+					}
+
+				case 'SHOW_SEARCH':
+					return function showSearch() {
+						self.showToolbarAtFront()
+						self.Store.setValue({ field: "searchActive", value: true });
+					}
+
+				default:
+					return null
+			}
+		}
+
+		/**
+		 * setHotkey
+		 *
+		 * This function does a some validation checks and then sets the global hotkey to the keyCombination with the action passed to it.
+		 * @param hotkey array of keys
+		 * @param action function to be assigned to the hotkey
+		 */
+		const setHotkey = ({ name, keyCombination, action, enabled }) => {
+			try {
+				const getAction = hotkeyActions(action)
+				// ensure the action is a function, the key combo is an array and not empty
+				if (typeof getAction === "function" && Array.isArray(keyCombination) && keyCombination.length && enabled) {
+					FSBL.Clients.HotkeyClient.addGlobalHotkey(keyCombination, getAction, () => { FSBL.Clients.Logger.log(`Hotkey added - ${name} ${keyCombination}`) })
+				} else {
+					throw 'The key combination or action supplied are not valid.'
+				}
+			} catch (err) {
+				FSBL.Clients.Logger.error(`Failed to add the hotkey ${keyCombination} for ${name}.`, err)
+			}
+		}
+
+		/**
+		 * updateDefaultHotkeysFromConfig
+		 *
+		 * This function takes the hotkeys from the config file and updates / merges the default hotkeys with it's values.
+		 * @param {Array} configHotkeys
+		 * @param {Array} defaultHotkeys
+		 */
+		const overrideDefaultHotkeysWithConfigHotkeys = (configHotkeys, defaultHotkeys) =>
+			defaultHotkeys.map(defaultHotkey => {
```

Comment:
This should probably be:
```suggestion
			defaultHotkeys.forEach(defaultHotkey => {
```

Reply:
Can you elaborate on the change from map to forEach? Is it a functional reason or is it not clear? 🙂

Reply:
@NinjaManatee ?

Reply:
@WatsonCIQ @amaroks `map` is for transforming one array into another and `forEach` is for looping over an array of objects. Since you aren't capturing the returned result of the mapping, I think a `forEach` is more correct. 

Upon reading this code again, I see where my confusion is. `overrideDefaultHotkeysWithConfigHotkeys` is a one line function, whose return is inferred to be the result of the mapping. For clarity, I think it would be good to add {} around the function and explicitly add the return.